eo-runtime/src/main/java/EOorg/EOeolang/EOrust.java

```diff
@@ -74,6 +76,11 @@ public class EOrust extends PhDefault {
      */
     private static final ConcurrentHashMap<String, String> NAMES;
 
+    /**
+     * All phis indexed while executing of native method.
+     */
+    private final Map<Integer, Phi> phis = new HashMap<>();
```

Comment:
@levBagryansky Should it be concurrent as well as `NAMES` ?

Reply:
@Graur This map is not a static so it will be created for every thread separately